src/sdltiles.cpp

```diff
@@ -2401,15 +2402,21 @@ const sound_effect* find_random_effect( const std::string &id, const std::string
 // Contains the chunks that have been dynamically created via do_pitch_shift. It is used to
 // distinguish between dynamically created chunks and static chunks (the later must not be freed).
 std::set<Mix_Chunk*> dynamic_chunks;
+
+std::mutex dynamic_chunks_mutex;
+
 // Deletes the dynamically created chunk (if such a chunk had been played).
 void cleanup_when_channel_finished( int channel )
 {
     Mix_Chunk *chunk = Mix_GetChunk( channel );
-    const auto iter = dynamic_chunks.find( chunk );
-    if( iter != dynamic_chunks.end() ) {
-        dynamic_chunks.erase( iter );
-        free( chunk->abuf );
-        free( chunk );
+    {
+        std::lock_guard<std::mutex> dyn_chunks_guard(dynamic_chunks_mutex);
+        const auto iter = dynamic_chunks.find(chunk);
+        if (iter != dynamic_chunks.end()) {
+            dynamic_chunks.erase(iter);
+            free(chunk->abuf);
+            free(chunk);
```

Comment:
To minimize the amount of time spent in the lock could modify to:
```c++
Mix_Chunk *chunk = Mix_GetChunk( channel );
bool removed = false;
{
    // Lock
    std::lock_guard<std::mutex> dyn_chunks_guard( dynamic_chunks_mutex );
    const auto iter = dynamic_chunks.find( chunk );
    if( iter != dynamic_chunks.end() ) {
        dynamic_chunks.erase( iter );
        removed = true;
    }
} // Release lock after minimum amount of work
if( removed ) {
    // Free chunk memory, does not require lock to prevent double-free because condition variable
    // is only set while in the lock.
    free( chunk->abuf );
    free( chunk );
}
```

Reply:
Thanks for the review - While I agree that moving the release to outside the lock scope would provide a performance benefit, I'm not sure if the decreased readability is worth the gain.

Reply:
Agreed, I don't think a short increase in how long the lock is held matters too much in this case.